### ISSUE TITLE
[slot_attention] Train and test subsets of CLEVRwithmasks used in Object Discovery task

### ISSUE THREAD

**Issue by vadimkantorov** [google-research/google-research#595]
@tkipf Could you please upload the file lists for CLEVR6 / CLEVR10 train and test splits?

From what I understood that the first 70K frames were used for training. How was the test split obtained? What was its size? Was filtering <= 6 objects done before or after getting the train/test splits?

Thank you!

**Comment by vadimkantorov** [comment]
I trained Object Discovery task, on regular full CLEVRtrain dataset. Evaluating ARI on my own sampled CLEVR6-like dataset from CLEVRwithmasks - getting ARI of only 92%.

Would it be possible to provide file lists for the train/test splits for the Object Discovery experiment?

cc @dosovits @tkipf

**Comment by tkipf** [comment]
Hi Vadim, thanks a lot for your comment. We used 320 examples for evaluation, which IIRC were taken from the split right after the first 70k examples which we used for training. I verified that the results did not vary significantly based on the evaluation split (i.e. always stayed in the 98-99% range; this holds even when evaluating on samples from the training set). We chose 320 evaluation examples to stay in line with IODINE, but to be safe I'd recommend taking a larger evaluation split in future experiments.

I think your discrepancy comes from training on a different dataset (the non-DeepMind version of CLEVR) and then trying to transfer to CLEVR w/ masks (i.e. DeepMind's version). We remarked in the Readme that this is not advisable since dataset statistics appear to differ between the two datasets (e.g. object scale, camera position etc.).

**Comment by vadimkantorov** [comment]
Do 70K count before or after filtering by <= 6 objects?

It would indeed be better to have the actual splits ^^ or at least a code snippet that does filtering / splits

FYI my pytorch reimplementation is at https://github.com/vadimkantorov/yet_another_pytorch_slot_attention - also there are some nuisance because of missing padding="same" mode in PyTorch, so I did some crude interpolate call instead of fighting the paddings.

**Comment by tkipf** [comment]
That's a great question! I just checked and we first filter by number of objects, which leaves about 50k examples, and then get the evaluation split (final 320 examples) from there and use the rest for training. Minor differences in this protocol should not matter much, however, as the variance between runs with different seeds on the same splits should be higher than what you get if you compared on different splits.
